gusto/initialisation_tools.py

```diff
@@ -279,3 +287,153 @@ def calculate_Pi0(state, theta0, rho0):
     Pi0 = assemble(Pi*dx)/assemble(Constant(1)*dx(domain=state.mesh))
 
     return Pi0
+
+
+def moist_hydrostatic_balance(state, theta_e, water_t, pi_boundary=Constant(1.0)):
+    """
+    Given a wet equivalent potential temperature, theta_e, and the total moisture
+    content, water_t, compute a hydrostatically balance virtual potential temperature,
+    dry density and water vapour profile.
+    :arg state: The :class:`State` object.
+    :arg theta_e: The initial wet equivalent potential temperature profile.
+    :arg water_t: The total water pseudo-mixing ratio profile.
+    :arg pi_boundary: the value of pi on the lower boundary of the domain.
+    """
+
+    theta0 = state.fields('theta')
+    rho0 = state.fields('rho')
+    water_v0 = state.fields('water_v')
+
+    # Calculate hydrostatic Pi
+    Vt = theta0.function_space()
+    Vr = rho0.function_space()
+    Vv = state.spaces("Vv")
+    n = FacetNormal(state.mesh)
+
+    param = state.parameters
+
+    # define some parameters as attributes
+    R_d = param.R_d
+    p_0 = param.p_0
+    cp = param.cp
+    c_pv = param.c_pv
+    c_pl = param.c_pl
+    R_v = param.R_v
+    L_v0 = param.L_v0
+    T_0 = param.T_0
+    w_sat1 = param.w_sat1
+    w_sat2 = param.w_sat2
+    w_sat3 = param.w_sat3
+    w_sat4 = param.w_sat4
+    g = param.g
+
+    quadrature_degree = (5, 5)
```

Comment:
Yes, that's better. The point is that you have hard coded a quadrature degree (because the one that Firedrake generates automatically is absurdly high).  This quadrature degree is presumably OK for the problems you're currently running but at some point in the future someone might run this code with higher order elements, in which case the quadrature degree you've hardcoded won't be sufficient. So what this does is grab the degree of the DG space and check if it is greater than 2 - this will trigger the warning.